### PR TITLE
fix (test): prepare for next version

### LINKED ISSUE #593: Use new @fastify org dependencies
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the issue has not already been raised

### Issue

Migrate to the @fastify org dependencies when Fastify v4 releases.

See https://github.com/fastify/fastify-swagger/pull/591

### PR DESCRIPTION
**Pull request by darkgl0w**
Hello.

This PR aims to :
- upgrade package dependencies
- update `@fastify/helmet` content security policy related tests to match the new defaults
- update `Joi` related tests to the new `Joi` API.
- update `swagger-ui-dist` to the latest version and properly trigger the creation of the static `swagger-initializer.js` file to avoid inline script execution
- migrate to `@fastify` scoped package dependencies
- update failing tests and make use of tap `equal` instead of deprecated `is`

closes #593
closes #596
closes #427

~~NB: current CI is green but Node.js 10 support has been dropped by some dependencies, so we should first drop Node.js 10 support before merging this.~~

#### Checklist

- [x] run `npm run test` ~~and `npm run benchmark`~~
- [x] tests and/or benchmarks are included
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by mcollina** [comment]
@darkgl0w any updates here?

**Comment by mcollina** [comment]
Is this ready to merge?

File: package.json [hunk 1]
```diff
@@ -41,26 +42,26 @@
...
   "homepage": "https://github.com/fastify/fastify-swagger#readme",
   "devDependencies": {
-    "@types/node": "^17.0.0",
-    "fastify": "^3.7.0",
-    "fastify-basic-auth": "^2.1.0",
-    "fastify-helmet": "^5.0.3",
+    "@types/node": "^17.0.13",
+    "fastify": "^3.27.0",
+    "fastify-basic-auth": "^2.2.0",
+    "fastify-helmet": "^7.0.1",
```
Comment (Fdawgs) [inline package.json:48]: Need replacing with `@fastify/basic-auth` and `@fastify/helmet`, and then any `requires()` of them.

**Review by Fdawgs — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (darkgl0w) [reply package.json]: Done.

**Comment by darkgl0w** [comment]
This is ready to merge if no changes are required during the review process ^^

**Review by Fdawgs — APPROVED** [review]

**Review by kibertoad — APPROVED** [review]

**Comment by mcollina** [comment]
Is this a semver-major change? Do we want it still based on Fastify v3, or should I bundle with Fastify v4?

**Comment by darkgl0w** [comment]
@mcollina > I didn't bump `@fastify/static` to version 6.0.0 so for now it works well with fastify v3.

Numbers are cheap we can bump a major (acting the drop of Node.js 10 and 12 support while keeping fastify v3 compatibility) and then bump another major when we land fastify v4 with a fastify requirement set to v4.x ^^

**Pull request merged**